api/v1/resourceset_types.go

```diff
@@ -158,18 +180,6 @@ func (in *ResourceSet) GetTimeout() time.Duration {
 	return timeout
 }
 
-// GetInputs returns the ResourceSet inputs array.
-func (in *ResourceSet) GetInputs() []ResourceSetInput {
```

Comment:
Instead of deleting this function we could actually put it to good use like you did for `(*ResourceSetInputProvider).GetDefaultInputs() (map[string]any, error)`.

We can move the code from here:

https://github.com/controlplaneio-fluxcd/flux-operator/blob/a06fccd81ace6c15343a35bb3a524c8dbcefa77d/internal/builder/resourceset.go#L26-L37

The signature would become:

```go
func (in *ResourceSet) GetInputs() ([]map[string]any, error)
```

In the current state this function doesn't have much value but if we parsed the JSONs then it would be useful.